subsys/shell/Kconfig

```diff
@@ -12,7 +12,7 @@ menu "Shell Options"
 
 menuconfig SHELL
 	bool "Enable shell"
-	select LOG_RUNTIME_FILTERING
+	imply LOG_RUNTIME_FILTERING
```

Comment:
Note that this means that `CONFIG_SHELL=y` + `CONFIG_LOG_RUNTIME_FILTERING=n` is a valid combo. `imply` does not prevent the user from turning stuff off manually.

Maybe that's fine, but thought I'd check.

Reply:
yes, that's how we want it. This feature (log runtime filtering) is very useful with ui (like shell) which can reconfigure logger in runtime but it's not obligatory. Some may want to not use it (e.g. to save space)

Reply:
Alright, sounds nice like a nice fit then.